### ISSUE TITLE
Create Supporter CPT

### ISSUE THREAD

**Issue by lovestulip** [OpenSourceOrg/dotOrg#149]
### Description

This request is to replicate the "Supporter" Custom Post Type (CPT) functionality from staging to production.

Ariel has tested the following code on the staging environment:

```
function display_cpt_supporters() {
    // Override the main query for this specific case
    query_posts(array(
        'post_type' => 'supporter',
        'posts_per_page' => -1,
        'orderby' => 'title',
        'order' => 'ASC'
    ));

    $output = '<div class="cpt-posts-container">';

    if (have_posts()) {
        while (have_posts()) {
            the_post();

            // Fetch ACF fields
            $name = get_field('name');
            $organization = get_field('organization');
            $quote = get_field('quote');
            $link = get_field('link');

            $output .= '<div class="cpt-post">';
            $output .= '<div class="supporter-info">';
            $output .= '<p><strong>' . esc_html($name) . '</strong><br />';

            // Check if the link field exists and is not empty
            if ($link) {
                $output .= '<i><a href="' . esc_url($link) . '" target="_blank">' . esc_html($organization) . '</a></i><br />';
            } else {
                $output .= '<i>' . esc_html($organization) . '</i><br />';
            }

            if ($quote) {
                $output .= '&quot;' . esc_html($quote) . '&quot;';
            }
            $output .= '</div>';
            $output .= '</div>';
        }
        wp_reset_query(); // Reset the query to the original
    } else {
        $output .= '';
    }

    $output .= '</div>';
    return $output;
}
add_shortcode('display_supporters', 'display_cpt_supporters');
```

Ariel has requested us to move this functionality to production and ensure it is safe to deploy without breaking existing functionality.

### Context

See ZD link on the internal dev request.

**Comment by gin0115** [comment]
Hey

This has been added to staging. I have it ready to go into production along with #152 

Also can confirm that just doing the export from ACF/SCF works for the post type and meta fields. Plus the WP export works fine for the post content too (i did this to test on my local)

If you would like me to importer those too, let me know :)

**Comment by lovestulip** [comment]
@ajolo2ky Can you please review[ the latest update ](https://github.com/OpenSourceOrg/dotOrg/issues/149#issuecomment-2511340212)and let us know how you'd like us to proceed?

**Comment by ajolo2ky** [comment]
Hi, thanks. Yes, please promote it to production.

Also, would it be possible in the wp-admin editing of the CPT, to remove the main body block and just show the custom fields?

<img width="1228" alt="image" src="https://github.com/user-attachments/assets/c1de1832-7185-436c-b917-35859a7df20c">

**Comment by gin0115** [comment]
I will get this pushed up now.

As for the content editor, you can remove that in the ACF Post Type settings
![image](https://github.com/user-attachments/assets/60a3ef41-f2ef-438a-aa57-25839b901ac6)
![image](https://github.com/user-attachments/assets/2c0848a2-2a75-4009-beb3-6680806c2f9d)

Do you want me to import the supporters that are present on the staging site? I have got the export file to hand (didn't share as this is a public repo)

**Comment by ajolo2ky** [comment]
Thanks! Yes, please import the ones that are already there.

**Comment by gin0115** [comment]
I will try and hop on the computer and get them imported 🙂

**Comment by gin0115** [comment]
This has all been imported, can we close this out now? 

@lovestulip | @ajolo2ky

**Comment by lovestulip** [comment]
@ajolo2ky Can you please confirm?

**Comment by ajolo2ky** [comment]
Hi, sorry for the delay. This is worked, but the slug of supporters appears to be /blog/supporter/[supporter], it should be /supporter/[supporter] without /blog. Is that feasible? thanks.

**Comment by lovestulip** [comment]
@ajolo2ky Let us know your thoughts on the above when you can! Thank you so much!

**Comment by gin0115** [comment]
Hi @ajolo2ky,  

Apologies for not being clearer earlier! The setting you're looking for is located under `Settings -> Permalinks`. This isn't an ACF-specific setting—it's a general WordPress configuration.  

I haven't made any changes to it since you currently have over 830 posts relying on that permalink structure (sorry for not checking this sooner).  

If we want supporters to have URLs that exclude the `/blog/`, we might need to make some different changes. As it stands, those URLs for supporters will show a blank page unless we add a template.

**Comment by ajolo2ky** [comment]
Yes, we don't want to mess the current permalink structure. Can we remove the individual pages created by this CPT? I think that'd be faster.

**Comment by gin0115** [comment]
Is the aim for the supporters only to be shown on a single page? if so we can remove the ability for supporters to have a `/supporter/joe-bloggs` url.

**Comment by ajolo2ky** [comment]
If we could have /supporter/ (instead of /blog/supporter), I'd prefer that. Having the unique URI for each supporter may have it's advantages. But if it's too problematic, for the time being I prefer just removing individual pages, just have them in [/ai/endorsements](https://opensource.org/ai/endorsements) and close this issue.

**Comment by xpurichan** [comment]
@gin0115 we'd also like to flag a related issue Ariel is experiencing, which is difficulty committing this PR:
https://github.com/OpenSourceOrg/dotOrg/pull/162

**Comment by gin0115** [comment]
Sorry not forgotten about this, im suddenly swamped with lots of issues and struggling getting through them.

Will reply for pr162 on there directly

**Comment by gin0115** [comment]
With this, it can be done using the acf settings 

![Image](https://github.com/user-attachments/assets/eb626fc2-9d78-49cc-8e7f-0cee5e8c2a46)

ive never used ACF to register post types, so didnt know it has this option.

This is done now, so links are not prefixed as `/blog/supporter`, now its just `/supporter/`

**Comment by gin0115** [comment]
#192 has now gone up and I have changed the permalinks (as above). Suspsect we can close this out now.

**Comment by lovestulip** [comment]
@ajolo2ky Please confirm if we can consider this issue resolved. Thanks!

**Comment by smaffulli** [comment]
we get an error when trying to submit the form:

[image: image.png]

**Comment by smaffulli** [comment]
I'm testing on this page: https://opensource.org/ai/endorsements

the issue happens with Firefox but not always on Firefox Private... this
may have something to do with ublock origin or something like that.

**Comment by gin0115** [comment]
Strange, i just submitted several submissions with different browsers and tried with ublock on or off.

**CHROME (Regular)**

Without 
```json
{
    "contact_form_id": 71702,
    "status": "mail_sent",
    "message": "Thank you for your message. It has been sent.",
    "posted_data_hash": "....",
    "into": "#wpcf7....",
    "invalid_fields": []
}
```

With
```json
{
    "contact_form_id": 71702,
    "status": "mail_sent",
    "message": "Thank you for your message. It has been sent.",
    "posted_data_hash": "....",
    "into": "#wpcf7-...",
    "invalid_fields": []
}
```

**FIREFOX (Regular)**
```json
{
    "contact_form_id":71702,
    "status":"mail_sent",
    "message":"Thank you for your message. It has been sent.",
    "posted_data_hash":"....",
    "into":"#wpcf7-....",
    "invalid_fields":[]
}
```

You don't have a pi hole or anything else running which could stop calls, do you? 

(Tests from a linux machine)

**Comment by smaffulli** [comment]
No, nothing. i'm not on a vpn or other filter.

We're receiving comments from users who have issues though, it's not just
me.

**Comment by gin0115** [comment]
it is a strange one. 

Do you want to open a new ticket for this, as it's not related to the supporter post type.

I just need to run some errands and I will look around online and see if anyone else is reporting this when I'm back.

**Comment by smaffulli** [comment]
I thought the issues were related. It's fine, we'll investigate this
further and open a separate ticket

**Comment by ajolo2ky** [comment]
Hi team, I'm closing this issue as the CPT part of it is finished and working. I've sent you a separate email to address the issue @smaffulli was flagging, with more in-depth detail of what I'm seeing under the hood.

Best,
